### ISSUE TITLE
valid ansible syntax, reported as problem.

### ISSUE THREAD

**Issue by piengeng** [ansible/ansible-lint#2607]
##### Summary
should not output a problem when it's not

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details
```yaml
- hosts: localhost
  become: true
  tasks:
    - name: dummy
      block:
        - { name: debug, ansible.builtin.debug: { msg: 'debug' } }
        - { name: fail, ansible.builtin.fail }
      become_user: username
      delegate_to: localhost

```

```console (paste below)
ansible --version
ansible [core 2.13.5rc1]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/henry/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /home/henry/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
  jinja version = 3.0.3
  libyaml = True
ansible-lint --version
ansible-lint 6.8.0 using ansible 2.13.5rc1
```

- ansible installation method: OS package
- ansible-lint installation method: pip (vscode)

##### OS / ENVIRONMENT
kubuntu jammy

##### STEPS TO REPRODUCE

above yaml

error message: `You can improve the task key order to: name, become_user, delegate_to, block`
```console (paste below)

```

##### Desired Behavior
no problem in vscode

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:

- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ajinkyau** [comment]
@piengeng It's a warning which suggests that we can improve the task key order by moving the `become_user` and `delegate_to` keys above the `block` keyword

something like:
```
---
- name: Testing
  hosts: localhost
  become: true
  tasks:
    - name: Dummy
      become_user: username
      delegate_to: localhost
      block:
        - { name: Debug, ansible.builtin.debug: { msg: "debug" } }
        - { name: Fail, ansible.builtin.fail }
```

**Comment by piengeng** [comment]
ah, thanks! the comments/error message doesn't make sense to me at all.
